module/zfs/dmu_send.c

```diff
@@ -62,21 +63,26 @@
 int zfs_send_corrupt_data = B_FALSE;
 int zfs_send_queue_length = 16 * 1024 * 1024;
 int zfs_recv_queue_length = 16 * 1024 * 1024;
+/* Set this tunable to FALSE to disable setting of DRR_FLAG_FREERECORDS */
+int zfs_send_set_freerecords_bit = B_TRUE;
```

Comment:
You may want to focus your attention on 81f588c.  The other patches are all cherry picked from OpenZFS with minimal changes for Linux.  It's only 81f588c which introduces the new libzfs core function and a new ZFS_IOC_SEND_NEW ioctl of the new style.

As for why the `zfs_send_set_freerecords_bit` tuning I'm not sure.  That was part of the original resumable recv patch and is a question for @ahrens.